kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -53,23 +53,27 @@ class KyuubiSessionImpl(
     case (key, value) => sessionConf.set(key, value)
   }
 
-  private val engine: EngineRef = EngineRef(sessionConf, user, handle)
+  private val engine: EngineRef = new EngineRef(sessionConf, user)
   private val sessionEvent = KyuubiSessionEvent(this)
   EventLoggingService.onEvent(sessionEvent)
 
   private var transport: TTransport = _
   private var client: KyuubiSyncThriftClient = _
 
+  private var _handle: SessionHandle = _
```

Comment:
```scala
  private final val _handle: SessionHandle = SessionHandle(protocol)
  override def handle: SessionHandle = _handle
```

This looks flaky and tricky, compared with `AbstractSession`, can you refine this?